MH-13103, Runtime UI NG JavaScript Dependencies

lkiesow opened the pull request:
This patch harmonizes the runtime-ui-ng module with other modules using
JavaScript by using npm as package manager for dependencies and ESLint
as linter.

lkiesow committed:
MH-13103, Runtime UI NG JavaScript Dependencies

This patch harmonizes the runtime-ui-ng module with other modules using
JavaScript by using npm as package manager for dependencies and ESLint
as linter.

lkiesow committed:
MH-13103, Runtime Info UI REST Docs Integration

Due to the build system change in the runtime-info-ui-ng module, jQuery
cannot be copied from there anymore and hence needs to be added as
direct dependency.

staubesv reviewed (changes requested):
This PR breaks the Runtime UI:

Before PR:

![screen shot 2018-09-17 at 18 47 00](https://user-images.githubusercontent.com/1590263/45638012-2bfde180-baac-11e8-9c12-08439919d8f2.png)

After PR:

![screen shot 2018-09-17 at 19 01 04](https://user-images.githubusercontent.com/1590263/45638016-3029ff00-baac-11e8-8b96-8f8c25e8f66a.png)

Tested with Chrome Version 69.0.3497.92 (Official Build) (64-bit)

Seems a jquery related issue:

Uncaught ReferenceError: $ is not defined
    at index.html:10

The browser reports a 404 on  http://localhost:8080/js/jquery/jquery.js

lkiesow commented:
Interesting. Seems like that page is using the jQuery delivered by the REST docs…
Will fix that ina moment.

staubesv commented on modules/runtime-info-ui/pom.xml:
```diff
@@ -136,6 +136,25 @@
+                <resource>
+                  <directory>node_modules/</directory>
+                  <includes>
+                    <include>jquery/dist/jquery.min.js</include>
```
Probably that means you need to adapt the index page of runtime-info-ui (not the rest_docs page)

(this is runtime-info-ui, not runtime-info-ui-ng)

lkiesow committed:
MH-13103, Runtime Info UI Index Page

This patch fixes the landing page of non-admin nodes. It now uses the
jQuery version of this bundle and serves its JavaScript in a separate
file.

staubesv commented on modules/runtime-info-ui/src/main/resources/ui/index.html:
```diff
@@ -5,104 +5,8 @@
-            });
-        });
-    </script>
+    <script type="text/javascript" src="js/jquery/dist/jquery.min.js"></script>
```
No strong opinion, but do we want the "dist" in the URL? "js/jquery.min.js" would be nicer ;-)

What do you think?

lkiesow replied:
We want. That makes the deployment much easier in case there are multiple libraries at one point. And really no one will care about this path ;-P

lkiesow committed:
MH-13103, Added newline rule

Adds a rule to ESLint to enforce unix linebreaks and as well as a
newline at the end of a file

staubesv commented:
Hi @lkiesow Just realised you added further rules in another. Please let me know when I shall continue the review.

lkiesow commented:
I'm done
